File: C#/Demo/Demo/WeatherDailyData.cs
using System;
using System.Collections.Generic;

namespace Library
{
    /**
     * The WeatherDailyData class represents hourly weather data. It encapsulates all relevant meteorological details
     * temperature, max temperature, min temperature, max feelslike, min feelslike, feelslike, humidity, dew, precip, precippprob,
     * precipcover, preciptype, snow, snowdepth, preciptype, windgust, windspeed, winddir, pressure, visibility, cloudcover,
     * solarradiation, solarenergy, uvindex, sunrise, sunriseEpoch, sunset, sunsetEpoch, moonphase, conditions
     * description,icon, stations, and source for a specific date.
     */
    public class WeatherDailyData
    {
        private DateTime datetime;
        private long? datetimeEpoch;
        private double? tempmax;
        private double? tempmin;
        private double? temp;
        private double? feelslikemax;
        private double? feelslikemin;
        private double? feelslike;
        private double? dew;
        private double? humidity;
        private double? precip;
        private double? precipprob;
        private double? precipcover;
        private List<string> preciptype;
        private double? snow;
        private double? snowdepth;
        private double? windgust;
        private double? windspeed;
        private double? winddir;
        private double? pressure;
        private double? cloudcover;
        private double? visibility;
        private double? solarradiation;
        private double? solarenergy;
        private double? uvindex;
        private string sunrise;
        private long? sunriseEpoch;
        private string sunset;
        private long? sunsetEpoch;
        private double? moonphase;
        private string conditions;
        private string description;
        private string icon;
        private List<string> stations;
        private string source;
        private List<Event> events;
        private List<WeatherHourlyData> weatherHourlyData;

        // Constructors
        public WeatherDailyData()
        {
            this.preciptype = new List<string>();
            this.stations = new List<string>();
            this.events = new List<Event>();
            this.weatherHourlyData = new List<WeatherHourlyData>();
        }

        // Getters and setters for each property
        public DateTime Datetime { get => datetime; set => datetime = value; }
        public long? DatetimeEpoch { get => datetimeEpoch; set => datetimeEpoch = value; }
        public double? TempMax { get => tempmax; set => tempmax = value; }
        public double? TempMin { get => tempmin; set => tempmin = value; }
        public double? Temp { get => temp; set => temp = value; }
        public double? FeelsLikeMax { get => feelslikemax; set => feelslikemax = value; }
        public double? FeelsLikeMin { get => feelslikemin; set => feelslikemin = value; }
        public double? FeelsLike { get => feelslike; set => feelslike = value; }
        public double? Dew { get => dew; set => dew = value; }
        public double? Humidity { get => humidity; set => humidity = value; }
        public double? Precip { get => precip; set => precip = value; }
        public double? PrecipProb { get => precipprob; set => precipprob = value; }
        public double? PrecipCover { get => precipcover; set => precipcover = value; }
        public List<string> PrecipType { get => preciptype; set => preciptype = value; }
        public double? Snow { get => snow; set => snow = value; }
        public double? SnowDepth { get => snowdepth; set => snowdepth = value; }
        public double? WindGust { get => windgust; set => windgust = value; }
        public double? WindSpeed { get => windspeed; set => windspeed = value; }
        public double? WindDir { get => winddir; set => winddir = value; }
        public double? Pressure { get => pressure; set => pressure = value; }
        public double? CloudCover { get => cloudcover; set => cloudcover = value; }
        public double? Visibility { get => visibility; set => visibility = value; }
        public double? SolarRadiation { get => solarradiation; set => solarradiation = value; }
        public double? SolarEnergy { get => solarenergy; set => solarenergy = value; }
        public double? UvIndex { get => uvindex; set => uvindex = value; }
        public string Sunrise { get => sunrise; set => sunrise = value; }
        public long? SunriseEpoch { get => sunriseEpoch; set => sunriseEpoch = value; }
        public string Sunset { get => sunset; set => sunset = value; }
        public long? SunsetEpoch { get => sunsetEpoch; set => sunsetEpoch = value; }
        public double? MoonPhase { get => moonphase; set => moonphase = value; }
        public string Conditions { get => conditions; set => conditions = value; }
        public string Description { get => description; set => description = value; }
        public string Icon { get => icon; set => icon = value; }
        public List<string> Stations { get => stations; set => stations = value; }
        public string Source { get => source; set => source = value; }
        public List<Event> Events { get => events; set => events = value; }
        public List<WeatherHourlyData> WeatherHourlyData { get => weatherHourlyData; set => weatherHourlyData = value; }
    }
}

File: Swift/Demo/Demo/WeatherDailyData.swift
import Foundation

/**
 * The WeatherDailyData class represents hourly weather data. It encapsulates all relevant meteorological details
 * temperature, max temperature, min temperature, max feelslike, min feelslike, feelslike, humidity, dew, precip, precipprob,
 * precipcover, preciptype, snow, snowdepth, windgust, windspeed, winddir, pressure, visibility, cloudcover,
 * solarradiation, solarenergy, uvindex, sunrise, sunriseEpoch, sunset, sunsetEpoch, moonphase, conditions,
 * description, icon, stations, and source for a specific date.
 */
class WeatherDailyData {
    private var _datetime: Date?
    private var _datetimeEpoch: Int64?
    private var _tempmax: Double?
    private var _tempmin: Double?
    private var _temp: Double?
    private var _feelslikemax: Double?
    private var _feelslikemin: Double?
    private var _feelslike: Double?
    private var _dew: Double?
    private var _humidity: Double?
    private var _precip: Double?
    private var _precipprob: Double?
    private var _precipcover: Double?
    private var _preciptype: [String]
    private var _snow: Double?
    private var _snowdepth: Double?
    private var _windgust: Double?
    private var _windspeed: Double?
    private var _winddir: Double?
    private var _pressure: Double?
    private var _cloudcover: Double?
    private var _visibility: Double?
    private var _solarradiation: Double?
    private var _solarenergy: Double?
    private var _uvindex: Double?
    private var _sunrise: String?
    private var _sunriseEpoch: Int64?
    private var _sunset: String?
    private var _sunsetEpoch: Int64?
    private var _moonphase: Double?
    private var _conditions: String?
    private var _description: String?
    private var _icon: String?
    private var _stations: [String]
    private var _source: String?
    private var _events: [Event]
    private var _weatherHourlyData: [WeatherHourlyData]

    // Constructors
    init() {
        self._preciptype = [String]()
        self._stations = [String]()
        self._events = [Event]()
        self._weatherHourlyData = [WeatherHourlyData]()
    }

    // Getters and setters for each property
    var _Datetime: Date? {
        get { return _datetime }
        set { _datetime = newValue }
    }
    var _DatetimeEpoch: Int64? {
        get { return _datetimeEpoch }
        set { _datetimeEpoch = newValue }
    }
    var _TempMax: Double? {
        get { return _tempmax }
        set { _tempmax = newValue }
    }
    var _TempMin: Double? {
        get { return _tempmin }
        set { _tempmin = newValue }
    }
    var _Temp: Double? {
        get { return _temp }
        set { _temp = newValue }
    }
    var _FeelsLikeMax: Double? {
        get { return _feelslikemax }
        set { _feelslikemax = newValue }
    }
    var _FeelsLikeMin: Double? {
        get { return _feelslikemin }
        set { _feelslikemin = newValue }
    }
    var _FeelsLike: Double? {
        get { return _feelslike }
        set { _feelslike = newValue }
    }
    var _Dew: Double? {
        get { return _dew }
        set { _dew = newValue }
    }
    var _Humidity: Double? {
        get { return _humidity }
        set { _humidity = newValue }
    }
    var _Precip: Double? {
        get { return _precip }
        set { _precip = newValue }
    }
    var _PrecipProb: Double? {
        get { return _precipprob }
        set { _precipprob = newValue }
    }
    var _PrecipCover: Double? {
        get { return _precipcover }
        set { _precipcover = newValue }
    }
    var _PrecipType: [String] {
        get { return _preciptype }
        set { _preciptype = newValue }
    }
    var _Snow: Double? {
        get { return _snow }
        set { _snow = newValue }
    }
    var _SnowDepth: Double? {
        get { return _snowdepth }
        set { _snowdepth = newValue }
    }
    var _WindGust: Double? {
        get { return _windgust }
        set { _windgust = newValue }
    }
    var _WindSpeed: Double? {
        get { return _windspeed }
        set { _windspeed = newValue }
    }
    var _WindDir: Double? {
        get { return _winddir }
        set { _winddir = newValue }
    }
    var _Pressure: Double? {
        get { return _pressure }
        set { _pressure = newValue }
    }
    var _CloudCover: Double? {
        get { return _cloudcover }
        set { _cloudcover = newValue }
    }
    var _Visibility: Double? {
        get { return _visibility }
        set { _visibility = newValue }
    }
    var _SolarRadiation: Double? {
        get { return _solarradiation }
        set { _solarradiation = newValue }
    }
    var _SolarEnergy: Double? {
        get { return _solarenergy }
        set { _solarenergy = newValue }
    }
    var _UvIndex: Double? {
        get { return _uvindex }
        set { _uvindex = newValue }
    }
    var _Sunrise: String? {
        get { return _sunrise }
        set { _sunrise = newValue }
    }
    var _SunriseEpoch: Int64? {
        get { return _sunriseEpoch }
        set { _sunriseEpoch = newValue }
    }
    var _Sunset: String? {
        get { return _sunset }
        set { _sunset = newValue }
    }
    var _SunsetEpoch: Int64? {
        get { return _sunsetEpoch }
        set { _sunsetEpoch = newValue }
    }
    var _MoonPhase: Double? {
        get { return _moonphase }
        set { _moonphase = newValue }
    }
    var _Conditions: String? {
        get { return _conditions }
        set { _conditions = newValue }
    }
    var _Description: String? {
        get { return _description }
        set { _description = newValue }
    }
    var _Icon: String? {
        get { return _icon }
        set { _icon = newValue }
    }
    var _Stations: [String] {
        get { return _stations }
        set { _stations = newValue }
    }
    var _Source: String? {
        get { return _source }
        set { _source = newValue }
    }
    var _Events: [Event] {
        get { return _events }
        set { _events = newValue }
    }
    var _WeatherHourlyData: [WeatherHourlyData] {
        get { return _weatherHourlyData }
        set { _weatherHourlyData = newValue }
    }
}
